iree/compiler/Conversion/LinalgToLLVM/LinalgTileAndVectorizePass.cpp

```diff
@@ -68,11 +68,24 @@ struct TileWorkgroups : public linalg::LinalgBaseTilingPattern {
 namespace {
 struct TileAndVectorizeWorkgroups
     : public PassWrapper<TileAndVectorizeWorkgroups, FunctionPass> {
+  TileAndVectorizeWorkgroups(bool vectorize = true)
```

Comment:
FYI, I am reworking tensor->vectors this will go away in favor of (tile-to-l1, tile-to-vectors, vectorize) passes

Reply:
Please dont split the passes :) . Its painful enough for the `workgroup` marker to survive pass boundaries. Splitting the passes this way is an unnecessary phase ordering issue.

Reply:
I don't see any difference between applying 3 consecutive list of patterns in once  pass vs 3 passes! In any case, the reason I split the patterns into 3 passes is tensor ops requires more canonicalization than we do in buffers here and I haven't find find a way to run canonicalization between patterns other than the painful by-by-op or when sometimes there is a utility method that collects canonicalizations patterns for specific dialect.

Reply:
It is different. There is a weird phase ordering between passes because of the markers. They are not independent passes that are self-contained. You need to set a marker in one pass, which is used in another pass. Its not recommended approach with good reason. The upstream Linalg CodegenStrategy too is a single pass. We have abused markers in our codegen pipeline. I am trying to roll that back.
W.R.T canonicalization patterns, we only have a subset of ops. We should just collect those patterns and run canonicalizations w.r.t to those. I dont think its that difficuly. In any case, this is not related to this PR, but I am going to insist on markers not crossing pass boundaries.

Reply:
There is CSE needed and other reasons for not to collect patterns, but I agree this isn't related to this PR sorry for cluttering comments here :p